src/Grpc/JsonTranscoding/src/Microsoft.AspNetCore.Grpc.JsonTranscoding/Internal/JsonRequestHelpers.cs

```diff
@@ -265,13 +276,39 @@ public static async Task WriteResponseMessage(HttpResponse response, Encoding en
             GrpcServerLog.ReceivedMessage(serverCallContext.Logger);
             return (TRequest)requestMessage;
         }
+        catch (JsonException ex)
+        {
+            GrpcServerLog.ErrorReadingMessage(serverCallContext.Logger, ex);
+            throw new RpcException(new Status(StatusCode.InvalidArgument, "Request JSON payload is not correctly formatted."));
+        }
         catch (Exception ex)
         {
             GrpcServerLog.ErrorReadingMessage(serverCallContext.Logger, ex);
-            throw;
+            throw new RpcException(new Status(StatusCode.InvalidArgument, ex.Message));
```

Comment:
Actually, are we okay with just throwing away the stack trace for random `Exception`s that we might not have thrown ourselves. The message might not be enough to determine where the exception originated, right? Also, it's a little weird to log an exception that isn't either the exception we throw or at least the `InnerException`.

Reply:
Thanks for pointing this out. Exception now captured on RpcException.Status.DebugException.